### PR TITLE
Nuzhy/trade parameters UI

### PR DESCRIPTION
**Pull request by Nuzhy-Deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/javascript/app/common/event_handler.js [hunk 1]
```diff
@@ -13,4 +13,29 @@ const onlyNumericOnKeypress = (ev, optional_value) => {
...
+    // Ensure only one decimal point is allowed
+    const decimalCount = (inputValue.match(/\./g) || []).length;
+    if (decimalCount > 1) {
+        inputValue = inputValue.replace(/\./g, (match, offset) => (offset === inputValue.indexOf('.') ? match : ''));
+    }
+
+    return inputValue;
+};
+
+module.exports = { onlyNumericOnKeypress, handleNumeric };
```
Comment (prince-deriv) [inline src/javascript/app/common/event_handler.js]: We have added `common/helpers` please move this there.

File: src/javascript/app/hooks/events.js [hunk 2]
```diff
@@ -74,3 +74,7 @@ export const triggerPurchaseChange = () => triggerCustomEvent('purchaseChange');
 // Trade Change
 export const useTradeChange = () => useCustomEvent('tradeChange');
 export const triggerTradeChange = () => triggerCustomEvent('tradeChange');
+
+// Barrier Change
+export const useBarrierChange = () => useCustomEvent('barrierChange');
+export const triggerBarrierChange = () => triggerCustomEvent('barrierChange');
```
Comment (prince-deriv) [inline src/javascript/app/hooks/events.js:80]: Do we still need this? Can we leverage on the existing script where barrierChange is being updated and pass the values using `tradeManager`

Comment (Nuzhy-Deriv) [reply src/javascript/app/hooks/events.js:80]: values are passed using tradeManager, but I separated the barrierChange to have the separate trigger

File: src/javascript/app/common/currency.js [hunk 3]
```diff
@@ -1,9 +1,14 @@
+const { default: tradeManager } = require('./trade_manager');
```
Comment (prince-deriv) [inline src/javascript/app/common/currency.js]:
```suggestion
const { tradeManager } = require('./trade_manager').default;
```

File: src/javascript/app/components/time_picker.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 const moment     = require('moment');
+const { default: tradeManager } = require('../common/trade_manager');
```
Comment (prince-deriv) [inline src/javascript/app/components/time_picker.js]:
```suggestion
const { tradeManager } = require('../common/trade_manager').default;
```

File: src/javascript/app/pages/trade/reset.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
 const CommonFunctions = require('../../../_common/common_functions');
+const { default: tradeManager } = require('../../common/trade_manager');
```
Comment (prince-deriv) [inline src/javascript/app/pages/trade/reset.js]:
```suggestion
const { tradeManager } = require('../../common/trade_manager').default
```

File: src/javascript/app/pages/trade/starttime.js [hunk 6]
```diff
@@ -111,19 +110,13 @@ const StartDates = (() => {
                 Dropdown('#date_start');
                 Defaults.set(DATE_START, target.value);
                 $('#time_start_row').setVisibility(target.value !== 'now');
-                // sessionStorage.setItem(
-                //     'start_dates',
-                //     JSON.stringify(start_dates_data)
-                // );
-                // triggerSessionChange();
```
Comment (prince-deriv) [inline src/javascript/app/pages/trade/starttime.js:118]: Can we remove?

File: src/javascript/app/pages/trade/tick.js [hunk 7]
```diff
@@ -2,6 +2,7 @@ const moment               = require('moment');
 const countDecimalPlaces   = require('./common_independent').countDecimalPlaces;
 const displayPriceMovement = require('./common_independent').displayPriceMovement;
 const underlyings          = require('./symbols').underlyings;
+const { default: tradeManager } = require('../../common/trade_manager');
```
Comment (prince-deriv) [inline src/javascript/app/pages/trade/tick.js]:
```suggestion
const { tradeManager } = require('../../common/trade_manager').default;
```

File: src/javascript/app/pages/trade/starttime.js (hunk 6 above)
Comment (Nuzhy-Deriv) [reply src/javascript/app/pages/trade/starttime.js:118]: this was old code which I removed.

**Review by prince-deriv — APPROVED** [review]

**Review by meenakshi-deriv — APPROVED** [review]

**Pull request merged**